st3/sublime_lib/pure_resource_path.py

```diff
@@ -0,0 +1,103 @@
+import posixpath
+from functools import total_ordering
+
+
+__all__ = ['PureResourcePath']
+
+
+@total_ordering
+class PureResourcePath():
+    """A representation of a Sublime Text resource path. Inspired by pathlib."""
+
+    def __init__(self, *pathsegments):
+        self._parts = tuple(
+            part
+            for segment in pathsegments if segment
+            for part in posixpath.normpath(str(segment)).split('/')
+        )
+
+    def __hash__(self):
+        return hash(self.parts)
+
+    def __repr__(self):
+        return "%s(%s)" % (type(self).__name__, repr(str(self)))
+
+    def __str__(self):
+        return '/'.join(self.parts)
+
+    def __eq__(self, other):
+        return self._parts == other.parts
+
+    def __lt__(self, other):
```

Comment:
Same as for `__eq__`, though I wonder whether we should implement this at all. I mean, how do you compare paths? Though, pathlib provides them, so we should follow that.

Reply:
Now raises `TypeError`.

Reply:
Done.